### ISSUE TITLE
Instructions Outdated

### ISSUE THREAD

**Issue by rrevanth** [pagespeed/ngx_pagespeed#1010]
Hi,

The instructions mentioned are outdated,we cannot download from the archive of github.It is throwing an error

**Comment by jeffkaufman** [comment]
Could you paste the exact commands your ran, along with their output?  It's working for me:

```
NPS_VERSION=1.9.32.6
wget https://github.com/pagespeed/ngx_pagespeed/archive/release-${NPS_VERSION}-beta.zip
unzip release-${NPS_VERSION}-beta.zip
cd ngx_pagespeed-release-${NPS_VERSION}-beta/
wget https://dl.google.com/dl/page-speed/psol/${NPS_VERSION}.tar.gz
```

https://developers.google.com/speed/pagespeed/module/build_ngx_pagespeed_from_source

**Comment by crowell** [comment]
@rrevanth if you're just copying and pasting to your shell (at least in my case with konsole + zsh + oh-my-zsh) `\` 
are automatically added to escape `{` like this
```bash
wget https://github.com/pagespeed/ngx_pagespeed/archive/release-$\{NPS_VERSION\}-beta.zip
```
so you'll have to manually remove the backslashes

**Comment by jeffkaufman** [comment]
@crowell do you know which of your fancy tools is doing the generally-helpful-but-not-what-we-want here magic?  If you pop up an xterm do you get it?

**Comment by crowell** [comment]
looks like  zsh does it for me (zsh with a blank config on xterm inserts the escape characters)

**Comment by rrevanth** [comment]
I pasted the exact commands,I am also using zsh as my command line.But the wget cannot download the files and I cannot download the files using downloader such as uget axel.I even tried to run the commands on bash

**Comment by jeffkaufman** [comment]
If you're using zsh, then as @crowell noticed escapes are getting added when you paste the commands.  You can either remove those backslashes manually so that what you paste looks like what you see on the instructions page, or you could run these commands where I've done the substitution for you:

```
cd
wget https://github.com/pagespeed/ngx_pagespeed/archive/release-1.9.32.6-beta.zip
unzip release-1.9.32.6-beta.zip
cd ngx_pagespeed-release-1.9.32.6-beta/
wget https://dl.google.com/dl/page-speed/psol/1.9.32.6.tar.gz
tar -xzvf 1.9.32.6.tar.gz  # extracts to psol/

cd
wget http://nginx.org/download/nginx-1.8.0.tar.gz
tar -xvzf nginx-1.8.0.tar.gz
cd nginx-1.8.0/
./configure --add-module=$HOME/ngx_pagespeed-release-1.9.32.6-beta
make
sudo make install
```

**Comment by rrevanth** [comment]
@jeffkaufman  yeah..zsh automatically adding slashes.It works by removing those.

Now i am getting in to this weird problem,upstream is not working if i install nginx according to this instructions.

Is this also problem with me or ?

**Comment by crowell** [comment]
@rrevanth can you paste your error log here https://gist.github.com/ so that we can see what the issue is?

**Comment by rrevanth** [comment]
@crowell actually there are no errors in the error log.But the website is showing only blank plane page.

**Comment by crowell** [comment]
can you give a link to the site, or give the html source when you add `?ModPagespeedFilters=+debug` to your query string?

**Comment by rrevanth** [comment]
I removed the nginx and installed old stable nginx from apt source .

I have no idea about ?ModPagespeedFilters=+debug . Can you give me any guide to enable upstream with nginx - pagespeed

**Comment by crowell** [comment]
as far as I'm aware pagespeed shouldn't interfere with using the upstream module (http://nginx.org/en/docs/http/ngx_http_upstream_module.html)

Can you try again with the debug query string, for example if your page is `example.com/index.html` do `example.com/index.html?PageSpeedFilters=+debug` and paste the output from the view source page.

**Comment by crowell** [comment]
also nginx's error log would be helpful here.
